CONTRIBUTING.md

```diff
@@ -245,4 +245,135 @@ If you're unsure where to add code (renderer, UI, extensions, plugins) see the b
 * `node_modules`: Location for Node.JS packages / dependencies.
 
 
-> **_NOTE:_** Each `extension` folder is a separately packaged module. If there are any issues with loading, make sure your module is packaged correctly.
\ No newline at end of file
+> **_NOTE:_** Each `extension` folder is a separately packaged module. If there are any issues with loading, make sure your module is packaged correctly.
+
+### Extensions
+
+Podman Desktop is organized so that you can modularly add new functionality in the form of "extensions" as well as the corresponding `extension-api`. This allows you to communicate with Podman Desktop without having to know the internal-workings. You look for the API call and Podman Desktop will do the rest.
+
+This is located in the `/extensions` folder.
+
+#### Creating a new extension
+
+When creating a new extension, import the extension API: `import * as extensionApi from '@tmpwip/extension-api';` All functionality with Podman Desktop is communicated through this API including registering the new extension. The API is located [here](https://github.com/containers/podman-desktop/blob/main/packages/extension-api/src/extension-api.d.ts).
+
+When loading an extension, Podman Desktop will:
+1. Search and load the JavaScript / TypeScript file specified in `main` entry of the `package.json` file in the extension directory (typically `extension.js`).
```

Comment:
it can't be TypeScript there, only JavaScript as it's the "result/output" that will be loaded by Node.js